aicsimageio/writers/ome_tiff_writer.py

```diff
@@ -0,0 +1,635 @@
+            T,C,Z,Y,X and optionally S
+        channel_names:
+            The names for each channel to be put into the OME metadata
+        image_name:
+            The name of the image to be put into the OME metadata
+        physical_pixel_sizes:
+            Z,Y, and X physical dimensions of each pixel,
+            defaulting to microns
+        channel_colors:
+            The channel colors to be put into the OME metadata
+        is_rgb:
+            is a S dimension present?  S is expected to be the last dim in
+            the data shape
+
+        Returns
+        -------
+        OME
+            An OME object that can be converted to a valid OME-XML string
+        """
+        num_images = len(data_shapes)
+        # resolve defaults that are None
+        if dimension_order is None:
+            dimension_order = [None] * num_images
+        if channel_names is None:
+            channel_names = [None] * num_images
+        if image_name is None:
+            image_name = [None] * num_images
+        if physical_pixel_sizes is None:
+            physical_pixel_sizes = [
+                types.PhysicalPixelSizes(1.0, 1.0, 1.0)
+            ] * num_images
+        if channel_colors is None:
+            channel_colors = [None] * num_images
+
+        # assert all lists are same length
+        if (
+            num_images != len(data_types)
+            or num_images != len(dimension_order)
+            or num_images != len(channel_names)
+            or num_images != len(image_name)
+            or num_images != len(physical_pixel_sizes)
+            or num_images != len(channel_colors)
+        ):
+            raise ValueError("Mismatched array counts in parameters to build_ome")
+
+        images = []
+        tiff_plane_offset = 0
+        for image_index in range(len(data_shapes)):
+            # correct the dimension_order for ome
+            ome_dimension_order, is_rgb = OmeTiffWriter._resolve_OME_dimension_order(
+                data_shapes[image_index], dimension_order[image_index]
+            )
+            img = OmeTiffWriter._build_ome_image(
+                image_index,
+                tiff_plane_offset,
+                data_shapes[image_index],
+                data_types[image_index],
+                is_rgb,
+                ome_dimension_order,
+                image_name[image_index],
+                physical_pixel_sizes[image_index],
+                channel_names[image_index],
+                channel_colors[image_index],
+            )
+            # increment tiff_plane_offset for next image
+            tiff_plane_offset += (
+                img.pixels.size_z * img.pixels.size_t * len(img.pixels.channels)
+            )
+            images.append(img)
+
+        ome_object = OME(creator=f"aicsimageio {get_module_version()}", images=images)
+
+        # validate! (TODO: Is there a better api in ome-types for this?)
+        test = to_xml(ome_object)
+        from_xml(test)
```

Comment:
Pinging @tlambert03, I would suspect there is likely a better way to do this but can't seem to find it in `ome-types` myself.

All this is trying to do is ensure that the OME object we manually created is valid to the spec.

Reply:
Correction. Sorry for pinging. I found it:

https://github.com/tlambert03/ome-types/blob/fb78d9c8e964a77d74d41ceb0b3dce75109f1753/src/ome_types/schema.py#L75

@toloudis  Looks like you can just dump to string then call `validate` haha.